src/pages/index.js

```diff
@@ -82,9 +47,187 @@ const addPlaceModal = new PopupWithForm(
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeReact
+  );
+  return card.getView();
+}
+
+/* GET Cards */
+let section;
+api
+  .getInitialCards()
+  .then((items) => {
+    section = new Section(
+      {
+        items: items,
+        renderer: (cardData) => {
+          const cardElement = getCardElement(cardData);
+          return cardElement;
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+/* GET Profile */
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+api
+  .getUserInfo()
+  .then((profileData) => {
+    userInfo.setUserInfo(profileData);
+  })
+  .catch((err) => console.error(err));
+
+/* PATCH Profile Edit Function */
+function handleProfileEditSubmit(profileInputValues) {
+  editProfileModal.setLoading(true);
+  const userData = {
+    name: profileInputValues.title,
+    about: profileInputValues.description,
+  };
+
+  api
+    .editUserInfo(userData.name, userData.about)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo(updatedUserData);
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      editProfileModal.setLoading(false);
+    });
+}
+
+/* PATCH Profile Avatar Function */
+function handleAvatarSubmit({ url }) {
+  updateAvatarModal.setLoading(true);
+
+  api
+    .updateAvatar(url)
+    .then((res) => {
+      userInfo.updateAvatar(url);
+      updateAvatarModal.reset();
+      updateAvatarModal.close();
+      console.log("Success:", res);
+    })
+    .catch((err) => {
+      console.error("Error updating profile avatar:", err);
+    })
+    .finally(() => {
+      updateAvatarModal.setLoading(true, "Save");
+    });
+}
+
+/* POST Add Place Function */
+function handleNewPlaceSubmit(placeCardData) {
+  function makeRequest() {
+    const cardData = {
+      name: placeCardData.title,
+      link: placeCardData.url,
+    };
+    return api
+      .addNewPlace(cardData.name, cardData.link)
+      .then((newPlaceCard) => {
+        const cardElement = getCardElement({
+          name: newPlaceCard.name,
+          link: newPlaceCard.link,
+          _id: newPlaceCard._id,
+        });
+        addPlaceModal.reset();
+        section.addItem(cardElement);
+      });
+  }
+  handleSubmit(makeRequest, addPlaceModal, "Saving...", "Save", () => {
+    addPlaceModal.setLoading(true, "Save");
+  });
+}
+
+//Like React//
+function handleLikeReact(cardId, likeStatus) {
+  if (likeStatus) {
+    return api.removeLikeReact(cardId);
+  } else {
+    return api.addLikeReact(cardId);
+  }
+}
+
+/* Edit Profile Button Listener */
+profileEditBtn.addEventListener("click", () => {
+  const { name, job } = userInfo.getUserInfo();
+  profileTitleInput.value = name;
+  profileDescriptionInput.value = job;
```

Comment:
**Could be improved**

Please, call `resetValidation` here for the profile opening to clear possible errors because you fill the inputs with valid data